projects/storefrontlib/src/cms-components/misc/icon/icon-loader.service.ts

```diff
@@ -1,14 +1,29 @@
 import { Injectable } from '@angular/core';
-import { IconConfig, ICON_TYPES } from './icon.model';
+import { WindowRef } from '@spartacus/core';
+import {
+  IconConfig,
+  IconConfigResource,
+  IconResourceType,
+  ICON_TYPE,
+} from './icon.model';
 
 @Injectable({
   providedIn: 'root',
 })
 export class IconLoaderService {
-  constructor(private config: IconConfig) {}
+  private loadedResources = [];
+  constructor(protected winRef: WindowRef, protected config: IconConfig) {}
 
-  useSvg(): boolean {
-    return this.config.icon && this.config.icon.useSvg;
+  /**
+   * Indicates whether the given icon type is configured to use SVG.
+   */
+  useSvg(iconType: ICON_TYPE): boolean {
+    return (
+      this.config.icon.resources &&
+      !!this.config.icon.resources.find(
+        res => res.types && res.types.includes(iconType)
```

Comment:
I might be crazy but should this be `res.types && res.type === IconResourceType.SVG && res.types.includes(iconType)` or is it assumed it's an SVG since there are types?

Reply:
yes, you're crazy ;) good spot though. I did it initially intentionally, but after a 2nd look I agree we check on it explicitly.  I'll also make the type mandatory for `IconConfigResource`.